File: LarkCardKit/Builders/CardLinkBuilder.cs
using LarkCardKit.Models;

namespace LarkCardKit.Builders;

public class CardLinkBuilder
{
    private readonly CardLink _cardLink = new();
    
    public CardLinkBuilder Url(string url)
    {
        _cardLink.Url = url;
        return this;
    }
    
    public CardLinkBuilder PcUrl(string pcUrl)
    {
        _cardLink.PcUrl = pcUrl;
        return this;
    }
    
    public CardLinkBuilder IosUrl(string iosUrl)
    {
        _cardLink.IosUrl = iosUrl;
        return this;
    }
    
    public CardLinkBuilder AndroidUrl(string androidUrl)
    {
        _cardLink.AndroidUrl = androidUrl;
        return this;
    }
    
    public CardLink Build() => _cardLink;
}

File: TypeScript/LarkCardKit/src/builders/CardLinkBuilder.ts
import type { CardLink } from '../models';

export class CardLinkBuilder {
  private _link: CardLink = { url: '' };

  url(url: string): this {
    this._link.url = url;
    return this;
  }

  pcUrl(url: string): this {
    this._link.pc_url = url;
    return this;
  }

  iosUrl(url: string): this {
    this._link.ios_url = url;
    return this;
  }

  androidUrl(url: string): this {
    this._link.android_url = url;
    return this;
  }

  build(): CardLink {
    return this._link;
  }
}
